### DISCUSSION TITLE
The incomprehensible behavior of the store in Firefox

### DISCUSSION THREAD

**Discussion by budarin** [pmndrs/zustand#2255] [Q&A] [answered]
The incomprehensible behavior of the store in Firefox

there is a store of the following structure:
```ts
{
  icons: {
     ids: [],
     byId: {}
  }
}
```
Adding new data to the store

```ts
const state = store.getState();
const icon =  { icon_id: 0, file_name: "IconName" };
const { icon_id } = icon;

const newState = {
    ...state,
    icons: {
        ...state.icons,
        ids: [...state.icons.ids, icon_id],
        byId: { ...state.icons.byId, [icon_id]: icon },
    },
};

store.setState(newState);

console.log('New store state', store.getState());
```
and see it in the console
```
{
  icons: {
     ids: [],
     byId: {}
  }
}
```
Everything works fine in Chrome!
What is the reason - I can not understand

**Comment by dbritto-dev** [comment]
@budarin would you mind sharing us a minimal repro on codesandbox? I would like to help

**Reply by budarin** [reply, marked as answer]
sorry - it was my mistake in middleware with startViewTransition: it presents in only in Chrome
